packages/appstore/src/components/add-more-wallets/wallet-add-card.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { TWalletInfo } from 'Types';
+import { Text, WalletCard } from '@deriv/components';
+import { useCurrencyConfig } from '@deriv/hooks';
+import { observer, useStore } from '@deriv/stores';
+import { getWalletCurrencyIcon } from 'Constants/utils';
+import wallet_description_mapper from 'Constants/wallet_description_mapper';
+
+type TAddWalletCard = {
+    wallet_info: React.PropsWithChildren<TWalletInfo>;
+};
+
+const AddWalletCard = observer(({ wallet_info }: TAddWalletCard) => {
+    const {
+        ui: { is_dark_mode_on },
+    } = useStore();
+
+    const { currency, landing_company_name, is_added } = wallet_info;
+    const { getConfig } = useCurrencyConfig();
+    const currency_config = getConfig(currency);
+    const crypto_currency = currency_config?.type === 'crypto';
+
+    const wallet_details = {
+        currency,
+        icon: crypto_currency ? getWalletCurrencyIcon(currency, is_dark_mode_on) : currency_config?.icon,
+        icon_type: crypto_currency ? 'crypto' : 'fiat',
```

Comment:
```suggestion
        icon: currency_config?.is_crypto ? getWalletCurrencyIcon(currency, is_dark_mode_on) : currency_config?.icon,
        icon_type: currency_config?.type,
```

Reply:
@farzin-deriv for me it doesn't make sense to use `getWalletCurrencyIcon` for `crypto` and and use `currency_config?.icon` for fiat, `getWalletCurrencyIcon` returns correct `icons` for `fiat` and `crypto`

Reply:
@sergei-deriv My suggestion here just was not to create a new variable `crypto_currency` since `useCurrencyConfig` already returns `is_crypto` to us.

Anyway, Regarding what you said, In my opinion, we should get rid of `getWalletCurrencyIcon` and handle the icon in the `useWalletsList` hook, similar to what we do in `useCurrencyConfig`. I think I saw someone did it in one of the PRs 🤔 Maybe it was one of your PR 🤔 If we do that then we won't need to have conditions in the code for the icon, We would just use `wallet.icon` or something like that 🚀

Reply:
I agree with @farzin-deriv we should address it for the next PR 😃 👍🏼